File: WWF/Hands on labs 3.0/VB/Lab02/Completed/Exercise3/CustomPropertySample/SendMailActivityLibrary/SendMailActivity.vb
'--------------------------------------------------------------------------------
' This file is a "Sample" as from Windows Workflow Foundation
' Hands on Labs RC
'
' Copyright (c) Microsoft Corporation. All rights reserved.
'
' This source code is intended only as a supplement to Microsoft
' Development Tools and/or on-line documentation.  See these other
' materials for detailed information regarding Microsoft code samples.
' 
' THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
' KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
' IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
' PARTICULAR PURPOSE.
'--------------------------------------------------------------------------------

Imports System.Net.Mail

'NOTE: When changing the namespace; please update XmlnsDefinitionAttribute in AssemblyInfo.vb
<ActivityValidator(GetType(ParametersValidator))> _
Public Class SendMailActivity
    Inherits System.Workflow.ComponentModel.Activity
    Public Sub New()
        MyBase.New()
        InitializeComponent()
    End Sub

    Public Shared MailToProperty As DependencyProperty = _
                DependencyProperty.Register("MailTo", GetType(String), GetType(SendMailActivity))

    <Description("To Email Address")> _
    <Category("EmailActivity")> _
    <Browsable(True)> _
    <DesignerSerializationVisibility(DesignerSerializationVisibility.Visible)> _
    Public Property MailTo() As String
        Get
            Return (CType((MyBase.GetValue(SendMailActivity.MailToProperty)), String))
        End Get
        Set(ByVal Value As String)
            MyBase.SetValue(SendMailActivity.MailToProperty, Value)
        End Set
    End Property

    Public Shared MailFromProperty As DependencyProperty = DependencyProperty.Register("MailFrom", GetType(String), GetType(SendMailActivity))

    <Description("From Email Address")> _
    <Category("EmailActivity")> _
    <Browsable(True)> _
    <DesignerSerializationVisibility(DesignerSerializationVisibility.Visible)> _
    Public Property MailFrom() As String
        Get
            Return (CType((MyBase.GetValue(SendMailActivity.MailFromProperty)), String))
        End Get
        Set(ByVal Value As String)
            MyBase.SetValue(SendMailActivity.MailFromProperty, Value)
        End Set
    End Property

    Public Shared SubjectProperty As DependencyProperty = DependencyProperty.Register("Subject", GetType(String), GetType(SendMailActivity))

    <Description("Subject of Email")> _
    <Category("EmailActivity")> _
    <Browsable(True)> _
    <DesignerSerializationVisibility(DesignerSerializationVisibility.Visible)> _
    Public Property Subject() As String
        Get
            Return (CType((MyBase.GetValue(SendMailActivity.SubjectProperty)), String))
        End Get
        Set(ByVal Value As String)
            MyBase.SetValue(SendMailActivity.SubjectProperty, Value)
        End Set
    End Property

    Public Shared BodyProperty As DependencyProperty = DependencyProperty.Register("Body", GetType(String), GetType(SendMailActivity))

    <Description("Body of Email")> _
    <Category("EmailActivity")> _
    <Browsable(True)> _
    <DesignerSerializationVisibility(DesignerSerializationVisibility.Visible)> _
    Public Property Body() As String
        Get
            Return (CType((MyBase.GetValue(SendMailActivity.BodyProperty)), String))
        End Get
        Set(ByVal Value As String)
            MyBase.SetValue(SendMailActivity.BodyProperty, Value)
        End Set
    End Property

    Protected Overrides Function Execute(ByVal executionContext As ActivityExecutionContext) _
                                As ActivityExecutionStatus
        Dim toAddress As MailAddress = New MailAddress(MailTo)
        Dim fromAddress As MailAddress = New MailAddress(MailFrom)

        Dim addresses As MailAddressCollection = New MailAddressCollection
        addresses.Add(toAddress)

        Dim msg As MailMessage = New MailMessage(fromAddress, toAddress)

        msg.Subject = Subject
        msg.Body = Body

        Dim mail As SmtpClient = New SmtpClient("localhost")

        mail.Send(msg)
        Return ActivityExecutionStatus.Closed
    End Function
End Class

File: WWF/Hands on labs 3.5/Lab02/CSharp/after/Exercise4/SendMailActivityLibrary/SendMailActivity.cs
//--------------------------------------------------------------------------------
// This file is a "Sample" as from Windows Workflow Foundation
// Hands on Labs RC
//
// Copyright (c) Microsoft Corporation. All rights reserved.
//
// This source code is intended only as a supplement to Microsoft
// Development Tools and/or on-line documentation.  See these other
// materials for detailed information regarding Microsoft code samples.
// 
// THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
// KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
// IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
// PARTICULAR PURPOSE.
//--------------------------------------------------------------------------------

using System.Net.Mail;
using System;
using System.ComponentModel;
using System.ComponentModel.Design;
using System.Collections;
using System.Drawing;
using System.Workflow.ComponentModel.Compiler;
using System.Workflow.ComponentModel.Serialization;
using System.Workflow.ComponentModel;
using System.Workflow.ComponentModel.Design;
using System.Workflow.Runtime;
using System.Workflow.Activities;
using System.Workflow.Activities.Rules;

namespace SendMailActivityLibrary
{
	[Designer(typeof(SendMailDesigner), typeof(IDesigner))]
	[ActivityValidator(typeof(ParametersValidator))]
	public partial class SendMailActivity : System.Workflow.ComponentModel.Activity
	{
		public SendMailActivity()
		{
			InitializeComponent();
		}
		public static DependencyProperty ToProperty = System.Workflow.ComponentModel.DependencyProperty.Register("To", typeof(string), typeof(SendMailActivity));

		[Description("To Email Address")]
		[Category("EmailActivity")]
		[Browsable(true)]
		[DesignerSerializationVisibility(DesignerSerializationVisibility.Visible)]
		public string To
		{
			get
			{
				return ((string)(base.GetValue(SendMailActivity.ToProperty)));
			}
			set
			{
				base.SetValue(SendMailActivity.ToProperty, value);
			}
		}

		public static DependencyProperty FromProperty = System.Workflow.ComponentModel.DependencyProperty.Register("From", typeof(string), typeof(SendMailActivity));

		[Description("From Email Address")]
		[Category("EmailActivity")]
		[Browsable(true)]
		[DesignerSerializationVisibility(DesignerSerializationVisibility.Visible)]
		public string From
		{
			get
			{
				return ((string)(base.GetValue(SendMailActivity.FromProperty)));
			}
			set
			{
				base.SetValue(SendMailActivity.FromProperty, value);
			}
		}

		public static DependencyProperty SubjectProperty = System.Workflow.ComponentModel.DependencyProperty.Register("Subject", typeof(string), typeof(SendMailActivity));

		[Description("Subject of Email")]
		[Category("EmailActivity")]
		[Browsable(true)]
		[DesignerSerializationVisibility(DesignerSerializationVisibility.Visible)]
		public string Subject
		{
			get
			{
				return ((string)(base.GetValue(SendMailActivity.SubjectProperty)));
			}
			set
			{
				base.SetValue(SendMailActivity.SubjectProperty, value);
			}
		}

		public static DependencyProperty BodyProperty = System.Workflow.ComponentModel.DependencyProperty.Register("Body", typeof(string), typeof(SendMailActivity));

		[Description("Body of Email")]
		[Category("EmailActivity")]
		[Browsable(true)]
		[DesignerSerializationVisibility(DesignerSerializationVisibility.Visible)]
		public string Body
		{
			get
			{
				return ((string)(base.GetValue(SendMailActivity.BodyProperty)));
			}
			set
			{
				base.SetValue(SendMailActivity.BodyProperty, value);
			}
		}

		protected override ActivityExecutionStatus Execute(ActivityExecutionContext executionContext)
		{

			MailAddress toAddress = new MailAddress(To);
			MailAddress fromAddress = new MailAddress(From);

			MailAddressCollection addresses = new MailAddressCollection();
			addresses.Add(toAddress);

			MailMessage msg = new MailMessage(fromAddress, toAddress);

			msg.Subject = Subject;
			msg.Body = Body;

			SmtpClient mail = new SmtpClient("localhost");

			mail.Send(msg);
			return ActivityExecutionStatus.Closed;

		}

	}
}
